src/licenseChecker.ts

```diff
@@ -0,0 +1,58 @@
+import { Message } from 'discord.js';
+import fetch from 'node-fetch';
+
+const ghRegex = /https?:\/\/github.com\/([\d\w-.]+\/[\d\w-.]+)/;
+const contentsRegex = /\{\+path\}/;
+
+const githubAPI = 'https://api.github.com/repos/';
+
+interface GitHubResponse {
+  value: {
+    content?: string;
+  };
+}
+
+const checkForLicense = async (message: Message) => {
+  if (message.content) {
+    const matches = message.content.match(ghRegex);
+    if (matches) {
+      const repo = await fetch(githubAPI + matches[1]).then((b) => b.json());
+      if (repo.license === null) {
+        const responses: GitHubResponse[] = <GitHubResponse[]>(
+          await Promise.allSettled([
+            fetch(repo.contents_url.replace(contentsRegex, 'README')).then(
+              (b) => b.json()
+            ),
+            fetch(repo.contents_url.replace(contentsRegex, 'README.md')).then(
+              (b) => b.json()
+            ),
+          ])
+        );
+
+        const isCopyrightAsserted = responses
+          .filter((r) => 'value' in r && 'content' in r.value)
+          .map((r) => {
+            const buffer = Buffer.from(r.value.content!, 'base64').toString(
```

Comment:
Currently you're filter doesn't narrow the type (it still assumes content is nullish even though we know by the time we get to `.map` it *has* to be present). Instead you can use a type predicate on the filter function return type to tell TS that, in fact, those properties are present, and remove the risky non-null assertion `!`.

```suggestion
        const isCopyrightAsserted = responses
          .filter((r): r is { value: { content: string } } => 'value' in r && 'content' in r.value)
          .map((r) => {
            const buffer = Buffer.from(r.value.content, 'base64').toString(
```